openslides_backend/action/actions/user/assign_meetings.py

```diff
@@ -0,0 +1,147 @@
+    CommitteeManagementLevel,
+    OrganizationManagementLevel,
+)
+from ....permissions.permission_helper import (
+    has_committee_management_level,
+    has_organization_management_level,
+    has_perm,
+)
+from ....permissions.permissions import Permissions
+from ....shared.exceptions import ActionException, PermissionDenied
+from ....shared.filters import And, FilterOperator, Or
+from ....shared.patterns import fqid_from_collection_and_id
+from ....shared.schema import id_list_schema
+from ...generics.update import UpdateAction
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionResultElement
+
+
+@register_action("user.assign_meetings")
+class UserAssignMeetings(UpdateAction):
+    """
+    Action to assign a user to multiple groups and meetings.
+    """
+
+    model = User()
+    schema = DefaultSchema(User()).get_update_schema(
+        additional_required_fields={
+            "meeting_ids": {
+                "description": "An array of meetings, in the user should be added to a group.",
+                **id_list_schema,
+            },
+            "group_name": {"type": "string"},
+        }
+    )
+    skip_archived_meeting_check = True
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        user_id = instance["id"]
+        meeting_ids = set(instance.pop("meeting_ids"))
+        group_name = instance.pop("group_name")
+
+        user = self.datastore.get(
+            fqid_from_collection_and_id("user", user_id),
+            [
+                "meeting_ids",
+                "group_$_ids",
+                *[f"group_${meeting_id}_ids" for meeting_id in meeting_ids],
+            ],
+        )
+        user_meeting_ids = set(user.get("meeting_ids", []))
+        filter_ = And(
+            FilterOperator("name", "=", group_name),
+            Or(
+                *[
+                    FilterOperator("meeting_id", "=", meeting_id)
+                    for meeting_id in meeting_ids
+                ]
+            ),
+        )
+        groups = self.datastore.filter("group", filter_, ["meeting_id", "user_ids"])
+        groups_meeting_ids = set(group["meeting_id"] for group in groups.values())
+        # Now split the meetings in the 3 categories
+        self.success = groups_meeting_ids
+        success_update = self.success.difference(user_meeting_ids)
+        self.standard_meeting_ids = meeting_ids.difference(
+            groups_meeting_ids
+        ).difference(user_meeting_ids)
+        self.nothing_meeting_ids = meeting_ids.difference(
+            groups_meeting_ids
+        ).intersection(user_meeting_ids)
+        # check if all meetings are set into one category
+        assert (
+            self.success.union(self.standard_meeting_ids).union(
+                self.nothing_meeting_ids
+            )
+            == meeting_ids
+        )
+
+        if not self.success:
```

Comment:
Maybe you want to use success_update instead of self.success and change the Message to something like f`Don't find a group with groupname  {group_name} to assign to in any meeting`

Reply:
Okay, use success_update instead of self.success and changed the error msg text.